src/api-v1.js

```diff
@@ -580,6 +580,197 @@ api.declare({
 
 api.declare({
   method: 'put',
+  route: '/ami-set/:id',
+  name: 'createAmiSet',
+  input: 'create-ami-set-request.json#',
+  deferAuth: true,
+  scopes: [['aws-provisioner:manage-ami-set:<amiSetId>']],
+  title: 'Create new AMI Set',
+  stability:  base.API.stability.stable,
+  description: [
+    'Create an AMI Set. An AMI Set is a collection of AMIs with a single name.',
+  ].join('\n'),
+}, async function (req, res) {
+  let input = req.body;
+  let id = req.params.id;
+
+  input.lastModified = new Date();
+
+  // Authenticate request with parameterized scope
+  if (!req.satisfies({id})) {
+    return;
+  }
+
+  // Create amiSet
+  let amiSet;
+  try {
+    amiSet = await this.AmiSet.create({
+      id: id,
+      amis: input.amis,
+      lastModified: new Date(),
+    });
+  } catch (err) {
+    // We only catch EntityAlreadyExists errors
+    if (!err || err.code !== 'EntityAlreadyExists') {
+      throw err;
+    }
+    amiSet = await this.AmiSet.load({id});
+
+    // Check if it matches the existing amiSet
+    let match = [
+      'id',
+      'amis',
+    ].every((key) => {
+      return _.isEqual(amiSet[key], input[key]);
```

Comment:
I can't remember off hand, but we should ensure that `_.isEqual` is a deep equality operator, since an amiSet is an object which collects other objects.